### ISSUE TITLE
Attribute sets not preserving attribute values when set is changed and attribute is placed in a diffrent group.

### ISSUE THREAD

**Issue by joshfedo** [magento/magento2#26870]
### Preconditions (*)
1. Magento ver. 2.3.3
2. PHP 7.1.33

### Steps to reproduce (*)
1. In the admin navigate to Stores -> Attribute Sets
2. Click on any attribute set (i'll use the default DBs "Gear" attribute set)
3.Move a attribute out of a group (ie. Search engine optimization -> url_key gets moved into Product Details -> url_key)
4. Edit a product (Catalog ->product -> Any product)
5. Take note that Search engine optimization -> url_key has a value set
6.Change the attribute set to our modified one ie. Gear
7. url_key is now blank

### Expected result (*)
1. url_key (or any shared attribute) preserves its value. (as it used to in older versions of magento 2)

### Actual result (*)
1. The attribute's input is now empty.

**Comment by joshfedo** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @joshfedo. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @joshfedo, here is your [Magento instance](https://i-26870-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26870-2-4-develop.instances.magento-community.engineering/admin_8a08](https://i-26870-2-4-develop.instances.magento-community.engineering/admin_8a08)
Login: `8d92b259` Password: `5f915275065c`
Instance will be terminated in up to 3 hours.

**Comment by joshfedo** [comment]
I can confirm this is still a issue on the magento 2.3.4 instance provided above.

**Comment by ajijshekh123** [comment]
Hello @joshfedo,
I am trying to reproduce this issue but I can't find it.
Can you please share me 1 video to reproduce this issue? 

Thanks.

**Comment by joshfedo** [comment]
@ajijshekh123 please review this video https://streamable.com/hivhg 

The video end abruptly but please note that the attribute I used in this example "url_key" is blank after changing the attribute set to "gear" at the end. 

I've also tested this on the test environment provided by creating new attribute sets and products and the bug persists.

**Comment by ajijshekh123** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ajijshekh123. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ajijshekh123, here is your [Magento instance](https://i-26870-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26870-2-4-develop.instances.magento-community.engineering/admin_d1b9](https://i-26870-2-4-develop.instances.magento-community.engineering/admin_d1b9)
Login: `a8ee1f98` Password: `28174747df0e`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @ajijshekh123 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-31514` were created

**Issue Available**: @ajijshekh123, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by ajijshekh123** [comment]
I have been reproduced this issue in the default Magento 2.4 instance.
See attached the video.
![Issue](https://user-images.githubusercontent.com/24711761/74505861-3b41c280-4f1e-11ea-8956-677f5415a38f.gif)

Thanks.

**Comment by joshfedo** [comment]
I would also like to note that **some** attribute will persist after you press save. Here is a additional screen recording where the attribute "short_description" does not persist https://streamable.com/zwp0o

**Comment by namthanh95** [comment]
@magento I am working on this

**Comment by SarojRam** [comment]
@magento I am working on this

**Comment by gabrieldagama** [comment]
@magento give me test instance

**Comment by gabrieldagama** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, here is your [Magento instance](https://i-26870-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26870-2-4-develop.instances.magento-community.engineering/admin_cb2e](https://i-26870-2-4-develop.instances.magento-community.engineering/admin_cb2e)
Login: `bb5ad135` Password: `e62d77eea7f1`
Instance will be terminated in up to 3 hours.

**Comment by baranada** [comment]
I can't reproduce this problem on current 2.4-develop branch.

First, I placed url_key, meta_keyword, meta_title, meta_discriptions attributes in Aset1, default group. and in Aset2, I moved them into seo group. And I set PRD1 to ASET1 and PRD2 for ASET2.
![](https://i.imgur.com/2vdzPMT.png)

I changed attribute set for PRD2 from ASET2 to ASET1. and data was preserved.
![](https://i.imgur.com/2vdzPMT.png)
